src/modules/widgets/saga.ts

```diff
@@ -546,10 +551,28 @@ export function* createWidget({
   }
 }
 
+export function* cloneWidget({
+  payload,
+}: ReturnType<typeof cloneWidgetAction>) {
+  const { widgetId } = payload;
+  const widgetSettings = yield select(getWidgetSettings, widgetId);
+  const clonedWidgetId = createWidgetId();
+  const state = yield select();
+  const widgetItem = getWidget(state, widgetId);
```

Comment:
To avoid 2x _yield_ - this could be replaced with:
```
const widgetItem = yield select(getWidget, widgetId);
```

What's more **widgetItem** contains widget settings so the:
```
const widgetSettings = yield select(getWidgetSettings, widgetId);
```
could be removed as settings could be accessed from `widgetItem`